addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -458,23 +462,20 @@ def _process_accrual_plans(self, date_to=False, force_period=False):
                 # We have to check for end of year actions if it is within our period
                 #  since we can create retroactive allocations.
                 if allocation.lastcall.year < allocation.nextcall.year and\
-                    (current_level.action_with_unused_accruals == 'lost' or\
-                    current_level.postpone_max_days > 0):
+                    current_level.action_with_unused_accruals == 'postponed' and\
+                    current_level.postpone_max_days > 0:
                     after_period_gains = allocation._process_accrual_plan_level(
                         current_level, period_start, allocation.nextcall + relativedelta(day=1, month=1),
                         period_end, allocation.nextcall)
-                    if current_level.action_with_unused_accruals == 'postponed':
-                        # Compute number of days kept
-                        allocation_days = allocation.number_of_days - allocation.leaves_taken
-                        allowed_to_keep = max(0, current_level.postpone_max_days - allocation_days)
-                        number_of_days = min(allocation_days, current_level.postpone_max_days)
-                        allocation.number_of_days = number_of_days + allocation.leaves_taken
-                        total_gained_days = sum(days_added_per_level.values())
-                        days_added_per_level.clear()
-                        days_added_per_level[current_level] = min(total_gained_days - after_period_gains, allowed_to_keep)
-                    else:
-                        allocation.number_of_days = allocation.leaves_taken
-                        days_added_per_level.clear()
+                    # Compute number of days kept
+                    allocation_days = allocation.number_of_days - allocation.leaves_taken
+                    allowed_to_keep = max(0, current_level.postpone_max_days - allocation_days)
+                    number_of_days = min(allocation_days, current_level.postpone_max_days)
+                    allocation.number_of_days = number_of_days + allocation.leaves_taken
+                    total_gained_days = sum(days_added_per_level.values())
+                    days_added_per_level.clear()
+                    days_added_per_level[current_level] = min(total_gained_days - after_period_gains, allowed_to_keep)
+
```

Comment:
This actually does not follow the logic of conserving the full period even if it is split between two separate years (after_period_gains = amount of days gained after the latest new year)
I guess we could run that code before the first call to `_process_accrual_plan_level`?

Reply:
True, I moved it before is this what you meant ?

Reply:
I think we don't need `after_period_gains` anymore, and `days_added_per_level[current_level]` should be `min(allowed_to_keep, total_gained_days)` I think

Reply:
yep totally, I forgot to remove it ^^